src/DataProtection/Abstractions/src/IDataProtector.cs

```diff
@@ -25,4 +27,30 @@ public interface IDataProtector : IDataProtectionProvider
         /// </exception>
         byte[] Unprotect(byte[] protectedData);
     }
+
+#if NETCOREAPP
+    /// <summary>
+    /// An interface that can provide data protection services using spans.
+    /// </summary>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0016:Add public types and members to the declared API", Justification = "<Pending>")]
+    public interface ISpanDataProtector : IDataProtectionProvider
+    {
+        /// <summary>
+        /// Cryptographically protects a piece of plaintext data.
+        /// </summary>
+        /// <param name="plaintext">The plaintext data to protect.</param>
+        /// <returns>The protected form of the plaintext data.</returns>
+        Span<byte> Protect(ReadOnlySpan<byte> plaintext);
```

Comment:
Usually you want to pass both the input and output spans to these APIs. So this would look like:

```C#
int Protect(ReadOnlySpan<byte> plainText, Span<byte> protectedData); // returns number of bytes written
```

Reply:
So the caller would be responsible for allocating larger arrays if its not big enough I assume?  So is it TryProtect where you return false if the span wasn't big enough?  Or we return int bytes or throw if the Span isn't large enough?